doc/dev_guide/c_style_guide_new.rst

```diff
@@ -0,0 +1,1046 @@
+    /** comes for documentation comments, /* for local not documented comments.
+    However the difference is vague already, so the rule is simple:
+    out of function = /\**, inside = /\*.
+
+    .. code-block:: c
+
+        /**
+         * Out of function comment, option 1.
+         */
+
+        /** Out of function comment, option 2. */
+
+        int
+        function()
+        {
+            /* Comment inside function, option 1. */
+
+            /*
+             * Comment inside function, option 2.
+             */
+        }
+
+    If a function has declaration and implementation separated, the function comment
+    should be for the declaration. Usually in the header file. Don't duplicate the
+    comment.
+
+    A comment and the function signature should be synchronized. Double-check if the
+    parameter names are the same as used in the comment, and mean the same.
+    Especially when you change one of them - ensure you changed the other.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Chapter 9: You've made a mess of it
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+That's OK, we all do.  You've probably been told by your long-time Unix
+user helper that ``GNU emacs`` automatically formats the C sources for
+you, and you've noticed that yes, it does do that, but the defaults it
+uses are less than desirable (in fact, they are worse than random
+typing - an infinite number of monkeys typing into GNU emacs would never
+make a good program).
+
+So, you can either get rid of GNU emacs, or change it to use saner
+values. To do the latter, you can stick the following in your .emacs file:
+
+.. code-block:: none
+
+  (defun c-lineup-arglist-tabs-only (ignored)
+    "Line up argument lists by tabs, not spaces"
+    (let* ((anchor (c-langelem-pos c-syntactic-element))
+           (column (c-langelem-2nd-pos c-syntactic-element))
+           (offset (- (1+ column) anchor))
+           (steps (floor offset c-basic-offset)))
+      (* (max steps 1)
+         c-basic-offset)))
+
+  (add-hook 'c-mode-common-hook
+            (lambda ()
+              ;; Add kernel style
+              (c-add-style
+               "linux-tabs-only"
+               '("linux" (c-offsets-alist
+                          (arglist-cont-nonempty
+                           c-lineup-gcc-asm-reg
+                           c-lineup-arglist-tabs-only))))))
+
+  (add-hook 'c-mode-hook
+            (lambda ()
+              (let ((filename (buffer-file-name)))
+                ;; Enable kernel mode for the appropriate files
+                (when (and filename
+                           (string-match (expand-file-name "~/src/linux-trees")
+                                         filename))
+                  (setq indent-tabs-mode t)
+                  (setq show-trailing-whitespace t)
+                  (c-set-style "linux-tabs-only")))))
+
+This will make emacs go better with the kernel coding style for C
+files below ``~/src/linux-trees``.
```

Comment:
This is not related to Tarantool. It assumes you have a special file with rules, and even if you really have it - it contains Linux rules, without our corrections. Another point why citing the kernel guide is a bad idea.

Reply:
Can I just remove this chapter or the part about `indent` below is ok?

Reply:
Looks irrelevant as well.